team/bundles/org.eclipse.compare/compare/org/eclipse/compare/internal/Utilities.java

```diff
@@ -179,7 +179,7 @@ public static void firePropertyChange(final ListenerList<IPropertyChangeListener
 			}
 		};
 		if (Display.getCurrent() == null) {
-			Display.getDefault().syncExec(runnable);
+			Display.getDefault().asyncExec(runnable);
 		} else {
 			runnable.run();
 		}
```

Comment:
What about leveraging the new `Display.getDefault().execute(runnable)`?

Reply:
ok